x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/test/integration/SingleNodeTests.java

```diff
@@ -41,28 +33,29 @@ protected boolean timeWarped() {
     // the index does not exist, a watch gets added
     // the watch should be executed properly, despite the index being created and the cluster state listener being reloaded
     public void testThatLoadingWithNonExistingIndexWorks() throws Exception {
-        stopWatcher();
-        ClusterStateResponse clusterStateResponse = client().admin().cluster().prepareState().get();
-        IndexMetaData metaData = WatchStoreUtils.getConcreteIndex(Watch.INDEX, clusterStateResponse.getState().metaData());
-        String watchIndexName = metaData.getIndex().getName();
-        assertAcked(client().admin().indices().prepareDelete(watchIndexName));
+        internalCluster().startNode();
+        ensureLicenseEnabled();
+        ensureWatcherTemplatesAdded();
+
+        assertFalse(client().admin().indices().prepareExists(Watch.INDEX).get().isExists());
         startWatcher();
 
         String watchId = randomAlphaOfLength(20);
         // now we start with an empty set up, store a watch and expected it to be executed
         PutWatchResponse putWatchResponse = watcherClient().preparePutWatch(watchId)
             .setSource(watchBuilder()
-                .trigger(schedule(interval(1, IntervalSchedule.Interval.Unit.SECONDS)))
+                .trigger(schedule(interval(1, IntervalSchedule.Interval.Unit.DAYS)))
                 .input(simpleInput())
                 .addAction("_logger", loggingAction("logging of watch _name")))
             .get();
         assertThat(putWatchResponse.isCreated(), is(true));
 
         assertBusy(() -> {
-            client().admin().indices().prepareRefresh(".watcher-history*");
-            SearchResponse searchResponse = client().prepareSearch(".watcher-history*").setSize(0).get();
-            assertThat(searchResponse.getHits().getTotalHits().value, is(greaterThanOrEqualTo(1L)));
-        }, 5, TimeUnit.SECONDS);
+            watcherClient().prepareExecuteWatch(watchId).setRecordExecution(true).get();
```

Comment:
by executing this manually, this does not check if the trigger mechanism is working as expected? Was this change intentional?

Reply:
I thought that the goal of the test was to test that Watcher set ups the indices correctly upon execution of a watch, whatever the execution mode is?

That's my understanding of the following comment:
  // the index does not exist, a watch gets added
  // the watch should be executed properly, despite the index being created and the cluster state listener being reloaded

Reply:
No, that means my comment is not useful and we have to fix it as well. The intention here is to make sure if a user adds the first watch without the index existing, that this watch gets added properly into the scheduling mechanism and executes. This used to be problematic when the index needed to be created first and could clear out the existing state.